src/components/chips/js/chipsController.js

```diff
@@ -0,0 +1,196 @@
+(function () {
+  'use strict';
+  angular
+      .module('material.components.chips')
+      .controller('MdChipsCtrl', MdChipsCtrl);
+
+
+
+  /**
+   * Controller for the MdChips component. Responsible for adding to and
+   * removing from the list of chips, marking chips as selected, and binding to
+   * the models of various input components.
+   *
+   * @param $mdUtil
+   * @param $mdConstant
+   * @param $log
+   * @ngInject
+   * @constructor
+   */
+  function MdChipsCtrl ($mdUtil, $mdConstant, $log) {
+    /** @type {Object} */
+    this.$mdConstant = $mdConstant;
+
+    /** @type {$log} */
+    this.$log = $log;
+
+    /** @type {angular.NgModelController} */
+    this.ngModelCtrl = null;
+
+    /** @type {Object} */
+    this.mdAutocompleteCtrl = null;
+
+    /** @type {Array.<Object>} */
+    this.items = [];
+
+    /** @type {number} */
+    this.selectedChip = -1;
+
+    /**
+     * Model used by the input element.
+     * @type {string}
+     */
+    this.chipBuffer = '';
+
+    /**
+     * Whether to use the mdChipAppend expression to transform the chip buffer
+     * before appending it to the list.
+     * @type {boolean}
+     */
+    this.useMdChipAppend = false;
+
+    /**
+     * Whether the Chip buffer is driven by an input element provided by the
+     * caller.
+     * @type {boolean}
+     */
+    this.hasInputElement = false;
+  }
+
+
+  /**
+   * Handles the keydown event on the input element: <enter> appends the
+   * buffer to the chip list, while backspace removes the last chip in the list
+   * if the current buffer is empty.
+   * @param event
+   */
+  MdChipsCtrl.prototype.workerInputKeydown = function(event) {
```

Comment:
The last chip in the list was labeled as a 'worker' chip, as a nod to the fact that it contains the input element. Since it's in the list of chips, it needs some special styling and so a distinguishing name. kept the CSS name, but switched to default in the js